packages/core/src/lib/history.unit.test.ts

```diff
@@ -0,0 +1,137 @@
+import { describe, expect, vi } from 'vitest';
+import { MINIMAL_CONFIG_MOCK } from '@code-pushup/test-utils';
+import { getCurrentBranchOrTag, safeCheckout } from '@code-pushup/utils';
+import { collectAndPersistReports } from './collect-and-persist';
+import { HistoryOptions, history, prepareHashes } from './history';
+import { upload } from './upload';
+
+vi.mock('@code-pushup/utils', async () => {
+  const utils: object = await vi.importActual('@code-pushup/utils');
+  let currentBranchOrTag = 'main';
+  return {
+    ...utils,
+    safeCheckout: vi.fn().mockImplementation((branch: string) => {
+      currentBranchOrTag = branch;
+    }),
+    getCurrentBranchOrTag: vi.fn().mockImplementation(() => currentBranchOrTag),
+  };
+});
```

Comment:
All tests are using `main` branch as the current branch.
Therefore, there is no need to implement additional logic for a potential different branch.

```suggestion
vi.mock('@code-pushup/utils', async () => {
  const utils: object = await vi.importActual('@code-pushup/utils');
  return {
    ...utils,
    safeCheckout: vi.fn(),
    getCurrentBranchOrTag: vi.fn().mockReturnValue('main'),
  };
});
});
```